recipe/meta.yaml

```diff
@@ -1,24 +1,24 @@
 {% set name = "gotrue" %}
-{% set version = "1.1.1" %}
+{% set version = "1.2.0" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 03c1593cff85027913bd1af063bcb38a5e79950fb5061768ff02ba7e67172708
+  sha256: f80befe60d713d5b524e70591fc22df4c5be5821d370585693cd76ac8c45eeeb
 
 build:
   number: 0
   noarch: python
-  script: {{ PYTHON }} -m pip install . -vv --no-deps
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
 
 requirements:
   host:
     - python >=3.8
     - pip
-    - poetry
+    - poetry <1.6
```

Comment:
According to the upstream repo, this should be `poetry-core >=1.0.0`
https://github.com/supabase-community/gotrue-py/blob/v1.2.0/pyproject.toml
I think that would work fine.  I think the issue you are seeing is just that the dependencies for recent versions of `poetry` aren't yet available for python 3.12 and you got sent back to a very old poetry that seems to be broken but which does happen to have all the dependencies build for python 3.12 right now. Given that you don't need `poetry`, just `poetry-core`, I think you'll save yourself a lot of pain by making that switch.